File: src/main/java/pl/alpheratzteam/obfuscator/core/transformer/SignatureTransformer.java
package pl.alpheratzteam.obfuscator.core.transformer;

import org.jetbrains.annotations.NotNull;
import org.objectweb.asm.tree.ClassNode;
import pl.alpheratzteam.obfuscator.api.transformer.Transformer;
import pl.alpheratzteam.obfuscator.api.util.StringUtil;

import java.util.Map;

/**
 * @author Unix on 07.10.2019.
 */
public class SignatureTransformer implements Transformer {
    @Override
    public void transform(@NotNull Map<String, ClassNode> classMap) {
        classMap.values().forEach(classNode -> classNode.signature = StringUtil.generateString(4));
    }

    @Override
    public String getName() {
        return "Signature";
    }
}

File: src/main/kotlin/pl/alpheratzteam/obfuscator/transformer/SignatureTransformer.kt
package pl.alpheratzteam.obfuscator.transformer

import pl.alpheratzteam.obfuscator.Obfuscator
import pl.alpheratzteam.obfuscator.api.transformer.Transformer
import pl.alpheratzteam.obfuscator.util.StringUtil

/**
 * @author Unix
 * @since 16.12.2020
 */

class SignatureTransformer : Transformer {
    override fun transform(obfuscator: Obfuscator) = obfuscator.classes.forEach { it.value.signature = StringUtil.makeUnreadable(StringUtil.generateString(4)) }
}